crates/parser/src/parse/statement_start.rs

```diff
@@ -191,25 +191,57 @@ pub static STATEMENT_START_TOKEN_MAPS: LazyLock<Vec<HashMap<SyntaxKind, Vec<Toke
             ],
         ));
 
-        // CREATE [ OR REPLACE ] OPERATOR
+        // CREATE OPERATOR
         m.push((
             SyntaxKind::DefineStmt,
             &[
                 SyntaxToken::Required(SyntaxKind::Create),
-                SyntaxToken::Optional(SyntaxKind::Or),
-                SyntaxToken::Optional(SyntaxKind::Replace),
                 SyntaxToken::Required(SyntaxKind::Operator),
             ],
         ));
 
-        // CREATE [ OR REPLACE ] TYPE
+        // CREATE TYPE name
         m.push((
             SyntaxKind::DefineStmt,
             &[
                 SyntaxToken::Required(SyntaxKind::Create),
-                SyntaxToken::Optional(SyntaxKind::Or),
-                SyntaxToken::Optional(SyntaxKind::Replace),
                 SyntaxToken::Required(SyntaxKind::TypeP),
+                SyntaxToken::Required(SyntaxKind::Ident),
```

Comment:
the fix part is here: `DefineStmt` was missing `SyntaxKind::Ident`

otherwise I have removed some illegal tokens for some statements (eg. `or replace`) based on the [docs][1]

[1]: https://neon.tech/docs/postgres/sql-createtype